src/lex.cpp

```diff
@@ -606,6 +606,16 @@ const char8* lexer::parse_string_literal() noexcept {
           ++c;
           break;
         }
+      case 'x':
+        for (int i = 0; i < 2; ++i) {
+          ++c;
+          if (!is_hex_digit(*c)) {
+            this->error_reporter_->report(error_invalid_hex_escape_sequence{
+                source_code_span(c-2-i, c+1)});
```

Comment:
Instead of computing `c-2-i`, store the start of the escape sequence in a variable outside the loop (e.g. after line 609).

Reply:
done, it made sense to me to place it slightly further up, let me know if that's no good